Fix: Add keep-alive options to Redis clients to prevent idle timeouts and socket closing.

nikitas-novatix opened the pull request:
**Problem:**

When Flowise is deployed with Redis (especially in Queue Mode or environments like Azure Cache for Redis with idle timeouts), Redis connections can be closed by the server after a period of inactivity (e.g., 10-15 minutes). This results in `uncaughtException: Socket closed unexpectedly` errors, crashing worker processes or causing failures in other Redis-dependent components.

This issue is tracked in #2186.

**Solution:**

This PR adds keep-alive mechanisms to the Redis clients used throughout Flowise to ensure connections remain active even during idle periods. The following settings were added to the respective client configurations:

-   **Core Queue Mode (@redis/client):** Added `socket: { keepAlive: 60000 }` and `pingInterval: 60000` to the clients used by `RedisEventPublisher` and `RedisEventSubscriber`.
-   **BullMQ (QueueManager - ioredis):** Added `keepAlive: 60000` and `enableReadyCheck: true` to the `ioredis` connection configuration used by BullMQ.
-   **Other `ioredis` Clients:** Added `keepAlive: 60000` and a default `retryStrategy` to clients used in `CachePool`, `RateLimiterManager`, `RedisCache`, `RedisEmbeddingsCache`, and `RedisBackedChatMemory` nodes.
-   **Other `@redis/client` Clients:** Added `socket: { keepAlive: 60000 }` and `pingInterval: 60000` to the client used in the `Redis` Vector Store node.

The keep-alive/ping intervals are set to 60000ms (1 minute) as a robust default, significantly shorter than typical server timeouts, ensuring connections are maintained without excessive background traffic.

**Testing:**

-   Successfully reproduced the "Socket closed unexpectedly" error locally using Queue Mode, SQLite, and a Dockerized Redis instance configured with a short (`--timeout 10`) idle timeout.
-   Verified that applying the keep-alive fixes with a short interval (e.g., 5000ms) prevented the original error under the same test conditions.

**Closes:**

Fixes #2186

HenryHengZJ commented on packages/components/nodes/cache/RedisCache/RedisCache.ts:
```diff
@@ -126,10 +126,15 @@ const getRedisClient = async (nodeData: INodeData, options: ICommonObject) => {
             password,
-            ...tlsOptions
+            ...tlsOptions,
+            keepAlive: 60000,
```
instead of hardcoding the values, can we have:

`REDIS_KEEP_ALIVE` env variable? `keepAlive: process.env.REDIS_KEEP_ALIVE ?? undefined,`

nikitas-novatix replied:
Yes, I'll change that now

nikitas-novatix commented on packages/components/nodes/cache/RedisCache/RedisCache.ts:
```diff
@@ -126,10 +126,15 @@ const getRedisClient = async (nodeData: INodeData, options: ICommonObject) => {
-        client = new Redis(redisUrl)
+        client = new Redis(redisUrl, {
+            keepAlive: 60000,
+            retryStrategy: (times) => Math.min(times * 100, 3000),
```
No, it was more like an enhancement to add some robustness. I'll remove it for consistency

nikitas-novatix commented:
@HenryHengZJ thanks for your comments.

I did the following changes:
- Added a REDIS_KEEP_ALIVE env variable, 60000 by default (1min).
- Added a one-liner based on your suggestion, but made it robust to mistakes in the environment variable. So if a person sets something that isn't a number, it still runs (without the KeepAlive mechanism)
- Simplified the code for consistency by removing non-necessary parts (pingInterval, RetryMechanism).
